articles/graph-traversals-java/index.md

```diff
@@ -0,0 +1,204 @@
+---
+
+title: Graph Implementation and Traversal Algorithms (Java)
+
+---
+Graphs are one of the most common data structures in computer science. Graphs are made up of nodes and edges. There are many applications for graph structures in the real world including relationships (Facebook), locations (Google Maps), programming analysis, and more.
+
+<!-- <img src="graph.png" alt="drawing" width="600"/> -->
+
+![graph](graph.jpg)
+
+By the end of this article, readers should know
+1. What graphs are
+2. How graphs are implemented in code
+3. How to traverse a graph
+    * Breadth first search iterative
+    * Depth first search iterative and recursive
+
+<br />
+
+## Types of graphs
+
+There are different types of graphs like undirected, directed, unweighted, unweighted, etc. All graphs have nodes and edges, but there are different structures and properties between different types. Undirected graphs have directionless edges between nodes while directed graphs have edges that point in a direction. Directed graphs can only be traversed in the direction the edges point. Unweighted graphs have zero value edges while weighted graphs have non zero value edges. These values can be positive or negative.
+
+<br />
+
+Undirected and Unweighted  |  Directed and Weighted	 |	
+:-------------------------:|:-----------------------:|
+|![undirected](undirected.png)|![directed](directed.png)
+
+<br />
+<br />
+
+## How to implement a graph
+
+### Node object
+
+First, we create a Node object. It contains a value and an array list of neighboring nodes.
+
+```
```

Comment:
Use the java syntax for formatting java code snippets.

Reply:
Hi, I'm not sure what you mean by that!